volatility/framework/plugins/windows/netscan.py

```diff
@@ -0,0 +1,305 @@
+        """Creates a list of Pool Tag Constraints for network objects.
+
+        Args:
+            context: The context to retrieve required elements (layers, symbol tables) from
+            symbol_table: The name of an existing symbol table containing the symbols / types
+
+        Returns:
+            The list containing the built constraints.
+        """
+
+        tcpl_size = context.symbol_space.get_type(symbol_table + constants.BANG + "_TCP_LISTENER").size
+        tcpe_size = context.symbol_space.get_type(symbol_table + constants.BANG + "_TCP_ENDPOINT").size
+        udpa_size = context.symbol_space.get_type(symbol_table + constants.BANG + "_UDP_ENDPOINT").size
+
+        # ~ vollog.debug("Using pool size constraints: TcpL {}, TcpE {}, UdpA {}".format(tcpl_size, tcpe_size, udpa_size))
+
+        return [
+            # TCP listener
+            poolscanner.PoolConstraint(b'TcpL',
+                                       type_name = symbol_table + constants.BANG + "_TCP_LISTENER",
+                                       size = (tcpl_size, None),
+                                       page_type = poolscanner.PoolType.NONPAGED | poolscanner.PoolType.FREE),
+            # TCP Endpoint
+            poolscanner.PoolConstraint(b'TcpE',
+                                       type_name = symbol_table + constants.BANG + "_TCP_ENDPOINT",
+                                       size = (tcpe_size, None),
+                                       page_type = poolscanner.PoolType.NONPAGED | poolscanner.PoolType.FREE),
+            # UDP Endpoint
+            poolscanner.PoolConstraint(b'UdpA',
+                                       type_name = symbol_table + constants.BANG + "_UDP_ENDPOINT",
+                                       size = (udpa_size, None),
+                                       page_type = poolscanner.PoolType.NONPAGED | poolscanner.PoolType.FREE)
+        ]
+
+    def determine_tcpip_version(self) -> str:
+        """Tries to determine which symbol filename to use for the image's tcpip driver. The logic is partially taken from the info plugin.
+
+        Args:
+
+        Returns:
+            The filename of the symbol table to use.
+        """
+
+        # while the failsafe way to determine the version of tcpip.sys would be to
+        # extract the driver and parse its PE header containing the versionstring,
+        # unfortunately that header is not guaranteed to persist within memory.
+        # therefore we determine the version based on the kernel version as testing
+        # with several windows versions has showed this to work out correctly.
+
+        is_64bit = symbols.symbol_table_is_64bit(self.context, self.config['nt_symbols'])
+
+        if is_64bit:
+            arch = "x64"
+        else:
+            arch = "x86"
+
+        # the following code is taken from the windows.info plugin.
```

Comment:
If this is identical code from `windows.info`, it might be worth us pulling that into a classmethod, and then you putting in a `PluginRequirement` and just calling that?

Reply:
yes, this would be great. Currently it's buried inside `windows.info`'s `_generator()`.

Reply:
The `windows.info` merge to master has happened, so hopefully you can make the updates now.  5:)

Reply:
done: https://github.com/volatilityfoundation/volatility3/pull/226/commits/cb5c234b06bc4aec1f5e63ed8fd586323cc98554